app/src/main/java/com/andrew/tasky/agenda/data/task/TaskDao.kt

```diff
@@ -1,3 +1,42 @@
 package com.andrew.tasky.agenda.data.task
 
-interface TaskDao
+import androidx.room.Delete
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import kotlinx.coroutines.flow.Flow
+
+interface TaskDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun upsertTask(task: TaskEntity): Long
+
+    @Query("SELECT * FROM TaskEntity WHERE time BETWEEN :startEpochMilli AND :endEpochMilli")
+    fun getTasksOfDateFlow(
+        startEpochMilli: Long,
+        endEpochMilli: Long
+    ): Flow<List<TaskEntity>>
+
+    @Query("SELECT * FROM TaskEntity WHERE time BETWEEN :startEpochMilli AND :endEpochMilli")
+    suspend fun getTasksOfDate(
+        startEpochMilli: Long,
+        endEpochMilli: Long
+    ): List<TaskEntity>
```

Comment:
It's enough to just have the flow function, the second one is equivalent to `flow.first()`